python/tests/test_map.py

```
@@ -494,6 +494,129 @@ def boom(_path):
+    class Dataset(DataArray):
+        def __init__(self, variables):
+            super().__init__()
+            self.data_vars = variables
+
+        def __getitem__(self, key):
+            return self.data_vars[key]
+
+        def __setitem__(self, key, value):
+            self.data_vars[key] = value
+
+        def copy(self):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `copy`

(ANN202)